### PR TITLE
refactor: use native token hook

### PR DESCRIPTION
**Pull request by hanszhoou**
## Description
Replace `useAsyncRetry()` by `useMemo()` since there is no async function is being adopted.

Closes # (NO_ISSUE)

## Type of change

- [ ] Documentation
- [ ] Code refactoring (Restructuring existing code w/o changing its observable behavior)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (a fix or feature that would make something no longer possible to do/require old user must upgrade their Mask Network to this new version)

## Checklist

- [ ] My code follows the style guidelines of this project.
- [ ] I have performed a self-review of my own code.
  - [ ] I have removed all in development `console.log`s
  - [ ] I have removed all commented code.
- [ ] I have commented on my code, particularly in hard-to-understand areas.
- [ ] I have read [Internationalization Guide](https://github.com/DimensionDev/Maskbook/blob/develop/docs/i18n-guide.md) and moved text fields to the i18n JSON file.

If this PR depends on external APIs:

- [ ] I have configured those APIs with CORS headers to let extension requests get passed. 
  - chrome extension: `chrome-extension://[id]`
  - firefox extension: `moz-extension://[id]`
- [ ] I have delegated all web requests to the background service via the internal RPC bridge.

### CODE REVIEW

File: packages/web3-shared/evm/hooks/useFungibleTokenDetailed.ts [hunk 1]
```diff
@@ -11,7 +12,8 @@ export function useFungibleTokenDetailed<
     address: string,
     token?: Partial<P>,
 ): AsyncStateRetry<P | undefined> {
-    const r1 = useNativeTokenDetailed()
+    const _r1 = useNativeTokenDetailed()
+    const r1 = useAsyncRetry(async () => _r1, [address, type, token])
```
Comment (UncleBill) [inline packages/web3-shared/evm/hooks/useFungibleTokenDetailed.ts]:
```suggestion
    const r1 = useNativeTokenDetailed()
    useEffect(r1.retry, [address, type, token])
```



But why `[address, type, token]` would affect to native token?

**Review by guanbinrui — APPROVED** [review]

**Review by guanbinrui — DISMISSED** [review]

**Comment by guanbinrui** [comment]
@zhouhanseng Plz resolve those conflicts.

**Review by UncleBill — DISMISSED** [review]
LGTM

**Review by guanbinrui — DISMISSED** [review]

**Comment by guanbinrui** [comment]
@zhouhanseng in conflicts.

**Pull request closed without merging**